maat-orchestration/src/main/java/uk/gov/justice/laa/crime/orchestration/service/CCLFUpdateService.java

```diff
@@ -68,14 +71,24 @@ public void updateSendToCCLF(WorkflowRequest request, RepOrderDTO repOrderDTO) {
             throw new ValidationException("Valid ApplicationDTO and RepOrderDTO is required");
         }
 
-        if (!compareRepOrderAndApplicationDTO(repOrderDTO, applicationDto)) {
-            SendToCCLFDTO sendToCCLFDTO = SendToCCLFDTO.builder().repId(repOrderDTO.getId())
-                    .applId(applicationDto.getApplicantDTO().getId())
-                    .applHistoryId(applicationDto.getApplicantDTO().getApplicantHistoryId()).build();
-            maatCourtDataApiService.updateSendToCCLF(sendToCCLFDTO);
+        if (featureDecisionService.isFeatureEnabled(request, FeatureToggle.MAAT_POST_ASSESSMENT_PROCESSING,
+                FeatureToggleAction.UPDATE)) {
+            if (!compareRepOrderAndApplicationDTO(repOrderDTO, applicationDto)) {
+                SendToCCLFDTO sendToCCLFDTO = SendToCCLFDTO.builder().repId(repOrderDTO.getId())
+                        .applId(applicationDto.getApplicantDTO().getId())
+                        .applHistoryId(applicationDto.getApplicantDTO().getApplicantHistoryId()).build();
+                maatCourtDataApiService.updateSendToCCLF(sendToCCLFDTO);
+            }
+        } else {
+            log.error("MAAT_POST_ASSESSMENT_PROCESSING feature is disabled {}", request.getUserDTO().getUserName());
```

Comment:
On reflection I actually think it would be preferable to move the feature toggle and `updateSendtoCCLF` call into the `ProceedingsService`, that way we can simply add it to the two existing methods, so if we add more workflows in the future we'll automatically get that functionality and won't have to repeat the logic in each `OrchestrationService`. It should also making mocking the feature toggle easier.

Here's what I'm thinking:

```java
public class ProceedingsService {

    private final CCLFUpdateService cclfUpdateService;
    private final ProceedingsMapper proceedingsMapper;
    private final ProceedingsApiService proceedingsApiService;

    public void updateApplication(WorkflowRequest request) {
        // Existing code
        updateCCLF(request);
    }

    public ApplicationDTO updateCrownCourt(WorkflowRequest request) {
        // Existing code
        updateCCLF(request);
    }

    public updateCCLF(WorkflowRequest request) {
        if (featureDecisionService.isFeatureEnabled(request, FeatureToggle.MAAT_POST_ASSESSMENT_PROCESSING,
                FeatureToggleAction.READ)) {
            cclfUpdateService.updateSendToCCLF(request);
        }
    }
```

Reply:
Updated